lib/recommendation-server.js

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const EXPORTED_SYMBOLS = ['RecommendationServer']; // eslint-disable-line no-unused-vars
+
+function RecommendationServer(opts) {
+  /*
+  Class providing a connection to a recommendation server and a method for
+  querying it. This class should be initialized once and exported as a
+  singleton to modules that will use it.
+
+  Usage:
+
+  let recommendation = new RecommendationServer({
+    // events is an instance of lib/events
+    events: events,
+    xhr: Cc['@mozilla.org/xmlextras/xmlhttprequest;1'],
+    server: 'http://universal-search.dev',
+    timeout: 1000
+  });
+  recommendation.init();
+  */
+  this.events = opts.events;
+  this.server = opts.server;
+  this.timeout = opts.timeout;
+  // note: xhr service, not xhr instance
+  this.xhr = opts.xhr;
+  this.console = opts.console;
+
+  this.search = this.search.bind(this);
```

Comment:
From IRC, for posterity:

    <_6a68> chuck: hey, what do you think of that PR branch? I'm not sure if it's solving the problem of app boilerplate, or really just half-finishing all the things. How's it look to you?
    <chuck> I wouldn’t have made all the same architectural decisions, but it’s all academic. If it works, let’s be pragmatic and move on.